frontend/src/components/project/ProjectOverviewChangeSection.tsx

```diff
@@ -20,33 +18,59 @@ const formSchema = z.object({
     .optional()
 });
 
-type FormData = z.infer<typeof formSchema>;
+const formSchemaWithSlug = baseFormSchema.extend({
+  slug: z
+    .string()
+    .min(1, "Required")
+    .max(64, "Too long, maximum length is 64 characters")
+    .regex(/^[a-zA-Z0-9-]+$/, "Only letters, numbers and hyphens are allowed")
+});
 
-export const ProjectOverviewChangeSection = () => {
+type BaseFormData = z.infer<typeof baseFormSchema>;
+type FormDataWithSlug = z.infer<typeof formSchemaWithSlug>;
+
+type Props = {
+  showSlugField?: boolean;
+};
+
+export const ProjectOverviewChangeSection = ({ showSlugField = false }: Props) => {
   const { currentWorkspace } = useWorkspace();
+  const [currentSlug, setCurrentSlug] = useState(currentWorkspace?.slug);
```

Comment:
Instead of manually monitoring the state change in react hook form, you can use watch provided by the library.

https://react-hook-form.com/docs/useform/watch